- predis -> 1.1.10 with ErrorResponse handling

vtl3000 opened the pull request:
- added exception handling to the pub/sub message loop
- removed job status check when setting job failed status

Hello!

* Type: bug fix | new feature | code quality | documentation
* Link to issue:

**In raising this pull request, I confirm the following (please check boxes):**

- [ ] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/cphalcon/blob/master/CONTRIBUTING.md)?
- [ ] I have checked that another pull request for this purpose does not exist.
- [ ] I wrote some tests for this PR.

Small description of change:

Thanks

Commit:
- predis -> 1.1.10 with ErrorResponse handling
- added exception handling to the pub/sub message loop

osadchyi-s commented on src/Jobs/BaseJob.php:
```diff
@@ -449,9 +449,9 @@ public function perform(): bool
-        if ($this->completed || $this->failed) {
-            throw new JobAlreadyFinishedException();
-        }
+//        if ($this->completed || $this->failed) {
```
Ми це додавали не просто так - там були трабли із цим серйозні

Slatch commented on src/Subscribers/WatchdogSubscriber.php:
```diff
@@ -105,58 +105,82 @@ public function watchdog(string $jid, string $worker, ?int $pid = null): void
+                ]
+            );
+            $this->logger->info(
+                "{type}: sending SIGKILL to child {$pid}; job {jid} canceled due to exception",
```
Тут, мабуть, так повинно бути?
```suggestion
                "{$type}: sending SIGKILL to child {$pid}; job {$jid} canceled due to exception",
```

serhii-nochevnyi replied:
там не змінні. в інших логах теж так.  власне звідти і копіював.

Slatch commented on composer.json:
```diff
@@ -31,7 +31,7 @@
         "ext-sockets": "*",
         "monolog/monolog": "^2.0",
-        "predis/predis": "1.1.*",
+        "predis/predis": "1.1.10",
```
чи не краще не завʼязуватись на конкретній версії, а додати, наприклад, хоча б `^`?

serhii-nochevnyi replied:
додав ^. та справа в тому, що потрібні зміни присутні лише в даній версії.

Commit:
- predis -> 1.1.10 with ErrorResponse handling
- added exception handling to the pub/sub message loop
- added monolog/monolog ^1.23 to backward compatibility

Commit:
- predis -> ^1.1.10 with ErrorResponse handling
- added exception handling to the pub/sub message loop
- added monolog/monolog ^1.23 to backward compatibility
